pkg/operator/resource/resourceapply/generic.go

```diff
@@ -52,6 +52,18 @@ type ApplyResult struct {
 	Error   error
 }
 
+// ConditionalFunction provides needed dependency for a resource on another condition instead of blindly creating
+// a resource. This conditional function can also be used to delete the resource when not needed
+type ConditionalFunction func() bool
+
+// ResourceConditionalMap maps a file to conditional functions that can be used to create or delete a resource in the
+// given file.
+type ResourceConditionalMap struct {
+	File                  string
+	DeleteConditionalFunc ConditionalFunction
+	CreateConditionalFunc ConditionalFunction
```

Comment:
What if both functions return `true`? IMO it would be better to have a single function that returns eg. `ResourceAction`, which can be either `"Create"` or `"Delete"`

Reply:
https://github.com/openshift/library-go/pull/1095/commits/59ec4d268990a31f4c0604ce2dc2c509ab5cac2c#diff-b91f253f2414046d3d35cc689c04410dfc00c7b8130a953ac59926118f9b8423R119 - Deletion trumps creation.

Reply:
Yes, I saw it. But what benefit do the two function have over a single function? IMO it makes the code more error prone.